third_party/move/tools/move-linter/src/model_ast_lints/cyclomatic_complexity.rs

```diff
@@ -0,0 +1,258 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! Cyclomatic complexity measures the number of independent execution paths
+//! through a function.  A high value generally correlates with code that is
+//! harder to test and maintain.
+//!
+//! This linter performs an approximation while traversing the Move
+//! expression tree:
+//!   1. The complexity score starts at **1**.
+//!   2. The score is incremented for each control-flow decision point found:
+//!         * +1 for each `if`
+//!         * +1 for each `else if`
+//!         * +1 for each `loop`, `while`, or `for`
+//!         * +1 for each `break` or `continue`
+//!         * +1 for each `return` statement that is not the final expression in the function
+//!         * +n where n = (number of match arms - 1)
+//!
+//! When the accumulated score exceeds `DEFAULT_THRESHOLD` (currently **10**),
+//! the linter emits a diagnostic suggesting that the function be simplified or
+//! decomposed.
+//!
+//! NOTE: The threshold is intentionally conservative.
+
+use move_compiler_v2::external_checks::ExpChecker;
+use move_model::{
+    ast::{ExpData, Operation, Value},
+    model::{FunctionEnv, Loc, NodeId},
+};
+
+const DEFAULT_THRESHOLD: i32 = 10;
+
+type FunKey = (move_model::model::ModuleId, move_model::model::FunId);
+
+#[derive(Default)]
+pub struct CyclomaticComplexity {
+    complexity: i32,
+    reported: bool,
+    root_node: Option<NodeId>,
+    current_fun: Option<FunKey>,
```

Comment:
Note that this struct is created once per every function. So no need to explicitly keep track of `current_fun`.

Also, you can implement your own `Default` for this struct (or a `new`, if you prefer that, and call that in `get_default_linter_pipeline`), which sets `complexity` to 1.